tools/shoestring/shoestring/internal/NodeKeyProvider.py

```diff
@@ -0,0 +1,25 @@
+import shutil
+from pathlib import Path
+
+
+class NodeKeyProvider:
```

Comment:
feel free to disregard, but i feel it would be cleaner to swap constructor and create parameters, e.g.

`NodeKeyProvider(factory).save_node_key(import_source=None)`

or, this could just be a standalone function:

`generate_node_private_key(factory, import_source=None)`

unlike `voter_configurator` and `harvester_configurator` it is only a single concern